### PR TITLE
RavenDB-22477 fixed issue with SearchEngineType being ignored for Counters and TimeSeries

### PR DESCRIPTION
**Pull request by ppekrol**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-22477

### Type of change

- [x] Bug fix
- [ ] Regression bug fix
- [ ] Optimization
- [ ] New feature

### How risky is the change?

- [x] Low 
- [ ] Moderate 
- [ ] High
- [ ] Not relevant

### Backward compatibility

- [x] Non breaking change
- [ ] Ensured. Please explain how has it been implemented?
- [ ] Breaking change
- [ ] Not relevant

### Is it platform specific issue?

- [ ] Yes. Please list the affected platforms.
- [x] No

### Documentation update

- [ ] This change requires a documentation update. Please mark the issue on YouTrack using `Documentation Required` tag.
- [x] No documentation update is needed 

### Testing by Contributor

- [x] Tests have been added that prove the fix is effective or that the feature works
- [ ] Internal classes added to the test class (e.g. entity or index definition classes) have the lowest possible access modifier (preferable `private`) 
- [ ] It has been verified by manual testing

### Testing by RavenDB QA team

- [ ] This change requires a special QA testing due to possible performance or resources usage implications (CPU, memory, IO). Please mark the issue on YouTrack using `QA Required` tag.
- [x] No special testing by RavenDB QA team is needed

### Is there any existing behavior change of other features due to this change?

- [ ] Yes. Please list the affected features/subsystems and provide appropriate explanation
- [x] No

### UI work

- [ ] It requires further work in the Studio. Please mark the issue on YouTrack using `Studio Required` tag.
- [x] No UI work is needed

### CODE REVIEW

**Review by arekpalinski — APPROVED** [review]

**Comment by garayx** [comment]
test this please

File: src/Raven.Client/Documents/Indexes/Counters/AbstractCountersIndexCreationTask.cs [hunk 1]
```diff
@@ -89,6 +89,11 @@ public override CountersIndexDefinition CreateIndexDefinition()
                     builder.AddMap(_map.Counter, _map.Map);
             }
 
+            if (SearchEngineType.HasValue)
+            {
```
Comment (maciejaszyk) [inline src/Raven.Client/Documents/Indexes/Counters/AbstractCountersIndexCreationTask.cs:93]: Are we missing `CompoundFieldsStrings` in the builder?

File: src/Raven.Client/Documents/Indexes/TimeSeries/AbstractTimeSeriesIndexCreationTask.cs [hunk 2]
```diff
@@ -89,6 +89,11 @@ public override TimeSeriesIndexDefinition CreateIndexDefinition()
                     builder.AddMap(_map.TimeSeries, _map.Map);
             }
 
+            if (SearchEngineType.HasValue)
+            {
+                builder.Configuration[Constants.Configuration.Indexes.IndexingStaticSearchEngineType] = SearchEngineType.Value.ToString();
+            }
```
Comment (maciejaszyk) [inline src/Raven.Client/Documents/Indexes/TimeSeries/AbstractTimeSeriesIndexCreationTask.cs:95]: Here as well

File: src/Raven.Client/Documents/Indexes/Counters/AbstractCountersIndexCreationTask.cs (hunk 1 above)
Comment (ppekrol) [reply src/Raven.Client/Documents/Indexes/Counters/AbstractCountersIndexCreationTask.cs:93]: Let's handle this as a separate issue

Comment (ppekrol) [reply src/Raven.Client/Documents/Indexes/Counters/AbstractCountersIndexCreationTask.cs:93]: https://issues.hibernatingrhinos.com/issue/RavenDB-22529/Add-CompoundFieldsStrings-to-Counter-and-TimeSeries-abstract-creation-tasks

**Pull request merged**